meson.build

```diff
@@ -62,18 +38,20 @@ else
     libtype = 'library'
 endif
 
+subdir('src')
```

Comment:
I'm a little less sure on the best way to make this work, but the ultimate problem I'm trying to solve is that when using nanoarrow as a subproject you end up with:

```sh
../subprojects/nanoarrow/src/nanoarrow/nanoarrow_types.h:24:10: fatal error: nanoarrow_config.h: No such file or directory
   24 | #include "nanoarrow_config.h"
      |          ^~~~~~~~~~~~~~~~~~~~
```

The CMake file configuration allows you to write directly to the source which is why that relative import works better. For Meson, we would have to either generate this header in `src/nanoarrow` and have the dependency include that path or generate in `src` as is done here, which the dependency already resolves includes to for supporting things like `#include nanoarrow/<>.h`

Another approach would be to change the include to `#include <nanoarrow/nanoarrow_config.h>` in nanoarrow_types.h and generate the configuration file in `src/nanoarrow`